libutempter/diag.h

```diff
@@ -45,12 +45,12 @@
 	} while (0)					\
 /* End of debug_msg definition.  */
 
-# define fatal_error(fmt_, ...)				\
-	do {						\
-		log_error(fmt_, ##__VA_ARGS__);		\
-		print_dbg(fmt_, ##__VA_ARGS__);		\
-		exit(EXIT_FAILURE);			\
-	} while (0)					\
+# define fatal_error(fmt_, ...)							\
+	do {									\
+		log_error("[ppid=%ld] " fmt_, (long)getppid(), ##__VA_ARGS__);	\
```

Comment:
Strictly speaking, the use of `getppid` requires adding `#include <unistd.h>` to `diag.h`.
However, in a tiny project like this I would probably not make it a strict requirement.